cmd/webhook/main.go

```diff
@@ -0,0 +1,133 @@
+
+	clientset "github.com/openebs/api/pkg/client/clientset/versioned"
+	//snapclientset "github.com/openebs/maya/pkg/client/generated/openebs.io/snapshot/v1alpha1/clientset/internalclientset"
+)
+
+var (
+	masterURL  string
+	kubeconfig string
+)
+
+func main() {
+	var parameters webhook.Parameters
+
+	// get command line parameters
+	flag.IntVar(&parameters.Port, "port", 8443, "Webhook server port.")
+	flag.StringVar(&parameters.CertFile, "tlsCertFile", "/etc/webhook/certs/cert.pem", "File containing the x509 Certificate for HTTPS.")
+	flag.StringVar(&parameters.KeyFile, "tlsKeyFile", "/etc/webhook/certs/key.pem", "File containing the x509 private key to --tlsCertFile.")
+
+	klog.InitFlags(nil)
+	err := flag.Set("logtostderr", "true")
+	if err != nil {
+		klog.Info(err, "failed to set logtostderr flag")
+	}
+	flag.Parse()
+
+	// Get in cluster config
+	cfg, err := getClusterConfig(kubeconfig)
+	if err != nil {
+		klog.Fatalf("Error building kubeconfig: %s", err.Error())
+	}
+	// Building Kubernetes Clientset
+	kubeClient, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		klog.Fatalf("Error building kubernetes clientset: %s", err.Error())
+	}
+	// Building OpenEBS Clientset
+	openebsClient, err := clientset.NewForConfig(cfg)
+	if err != nil {
+		klog.Fatalf("Error building openebs clientset: %s", err.Error())
+	}
+
+	// Building Snapshot Clientset
+	// snapClient, err := snapclientset.NewForConfig(cfg)
+	// if err != nil {
+	// 	klog.Fatalf("Error building openebs snapshot clientset: %s", err.Error())
+	// }
+
+	// Fetch a reference to the admission server deployment object
+	ownerReference, err := webhook.GetAdmissionReference(kubeClient)
+	if err != nil {
+		klog.Fatal(err, "failed to get a reference to the admission deployment object")
+	}
+	validatorErr := webhook.InitValidationServer(*ownerReference, kubeClient)
+	if validatorErr != nil {
+		klog.Fatal(validatorErr, "failed to initialize validation server")
+	}
+
+	wh, err := webhook.New(parameters, kubeClient, openebsClient)
+	if err != nil {
+		klog.Fatalf("failed to create validation webhook: %s", err.Error())
+	}
+	// define http server and server handler
+	mux := http.NewServeMux()
+	mux.HandleFunc("/validate", wh.Serve)
+	wh.Server.Handler = mux
+
+	// start webhook server in new routine
+	go func() {
+		if err := wh.Server.ListenAndServeTLS("", ""); err != nil {
+			klog.Errorf("Failed to listen and serve webhook server: %v", err)
+		}
+	}()
+
+	klog.Info("Webhook server started")
+
+	// listening OS shutdown singal
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, syscall.SIGINT, syscall.SIGKILL, syscall.SIGTERM)
+	<-signalChan
+
+	klog.Infof("Got OS shutdown signal, shutting down webhook server gracefully...")
+	err = wh.Server.Shutdown(context.Background())
+	if err != nil {
+		klog.Errorf("failed to shutdown server: error {%v}", err)
+	}
+}
+
+// GetClusterConfig return the config for k8s.
+func getClusterConfig(kubeconfig string) (*rest.Config, error) {
```

Comment:
this func is getting used every where let's have in a single place and use it instead , we can do this change later in upcoming PRs.
Better to use below func helpful in case of outofcluster runs as binary 
```go
// GetClusterConfig return the config for k8s.
func getClusterConfig(kubeconfig string) (*rest.Config, error) {

	if kubeconfig != "" {
		return clientcmd.BuildConfigFromFlags("", kubeconfig)
	}
	return rest.InClusterConfig()
}

```

Reply:
Above change can be done later.